src/git.ts

```diff
@@ -0,0 +1,1219 @@
+
+    // Compute an absolute path. If the stageFiles is already absolute, use it.
+    // Otherwise, join it with the repository's root path.
+    let absolutePath: string;
+    if (path.isAbsolute(stageFiles)) {
+        absolutePath = stageFiles;
+    } else {
+        absolutePath = path.join(repo.rootUri.fsPath, stageFiles);
+    }
+
+    // Pass the absolute path to the Git add command.
+    repo.add([absolutePath]).then(() => {
+        NEURO.client?.sendContext(`Added ${stageFiles} to staging area.`);
+    }, (err: string) => {
+        NEURO.client?.sendContext(`Adding files to staging area failed`);
+        logOutput("ERROR", `Failed to git add: ${err}`)
+    });
+}
+
+export function handleGitRemove(actionData: any) {
+    if (!git) {
+        NEURO.client?.sendActionResult(actionData.id, true, 'Git extension not available.');
+        return;
+    }
+    if (!vscode.workspace.getConfiguration('neuropilot').get('permission.gitOperations', false)) {
+        NEURO.client?.sendActionResult(actionData.id, true, 'You do not have permission to perform Git operations.');
+        return;
+    }
+    if (!repo) {
+        NEURO.client?.sendActionResult(actionData.id, true, 'You are not in a git repository.');
+        return;
+    }
+
+    
+    const filePath: string = actionData.params?.filePath;
+
+    NEURO.client?.sendActionResult(actionData.id, true)
+
+    // Normalize the file path if provided; otherwise, use wildcard.
+    const revertFiles: string = filePath ? getNormalizedRepoPathForGit(filePath) : `*`;
+
+    // Compute an absolute path. If the removeFiles is already absolute, use it.
+    // Otherwise, join it with the repository's root path.
+    let absolutePath: string;
+    if (path.isAbsolute(revertFiles)) {
+        absolutePath = revertFiles;
+    } else {
+        absolutePath = path.join(repo.rootUri.fsPath, revertFiles);
+    }
+
+    // Pass the absolute path to the Git remove command.
+    repo.revert([absolutePath]).then(() => {
+        NEURO.client?.sendContext(`Removed ${revertFiles} from the index.`);
+    }, (err: string) => {
+        NEURO.client?.sendContext(`Reverting files failed`);
+        logOutput("ERROR", `Git revert failed: ${err}`)
+    });
+}
+
+export function handleGitCommit(actionData: any) {
+    if (!git) {
+        NEURO.client?.sendActionResult(actionData.id, true, 'Git extension not available.');
+        return;
+    }
+    if (!vscode.workspace.getConfiguration('neuropilot').get('permission.gitOperations', false)) {
+        NEURO.client?.sendActionResult(actionData.id, true, 'You do not have permission to perform Git operations.');
+        return;
+    }
+    if (!repo) {
+        NEURO.client?.sendActionResult(actionData.id, true, 'You are not in a git repository.');
+        return;
+    }
+
+    
+    const message = `Neuro commit: ${actionData.params?.message}`;
+    const commitOptions: string[] | undefined = actionData.params?.commitOptions;
```

Comment:
Forgot about this in the last review.
```suggestion
    const commitOptions: string[] | undefined = actionData.params?.options;
```